components/resourceDetails/resourceSharing/addAgentButton/agentPickerModal/index.jsx

```diff
@@ -58,33 +47,72 @@ import AddAgentRow from "../addAgentRow";
 import AgentPickerEmptyState from "../agentPickerEmptyState";
 import styles from "./styles";
 import AddWebIdButton from "./addWebIdButton";
+import WebIdCheckbox from "./webIdCheckbox";
 import ConfirmationDialogContext from "../../../../../src/contexts/confirmationDialogContext";
 import useNamedPolicyPermissions from "../../../../../src/hooks/useNamedPolicyPermissions";
 import usePermissionsWithProfiles from "../../../../../src/hooks/usePermissionsWithProfiles";
+import useContacts from "../../../../../src/hooks/useContacts";
+import { GROUP_CONTACT } from "../../../../../src/models/contact/group";
+import {
+  PERSON_CONTACT,
+  findPersonContactInAddressBook,
+  savePerson,
+} from "../../../../../src/models/contact/person";
+import useAddressBook from "../../../../../src/hooks/useAddressBook";
+
+function serializePromises(promiseFactories, postResolveActions) {
```

Comment:
This smells like a reusable function - maybe move it to a utils module? We only have `src/solidClientHelpers/utils.js` till now, but we could introduce a promise model that has these utility functions involving promises.

Reply:
Let's move postResolveActions out - Let's avoid callbacks if possible (i.e. just use `await serializePromises` and then list the actions afterwards).